### PR TITLE
libsel4platsupport: check num_pages after alloc success

### PR DESCRIPTION
**Pull request by anchao**
libsel4platsupport: check num_pages after alloc success

To avoid null pointer reference

### CODE REVIEW

File: libsel4platsupport/src/io.c [hunk 1]
```diff
@@ -141,11 +141,11 @@ static void *sel4platsupport_map_paddr_with_page_size(sel4platsupport_io_mapper_
     size += offset;
 
     io_mapping_t *mapping = new_node(BYTES_TO_SIZE_BITS_PAGES(size, page_size_bits));
-    assert(mapping->num_pages << page_size_bits >= size);
     if (!mapping) {
-        ZF_LOGE("Failed to allocate node for %zu pages", mapping->num_pages);
+        ZF_LOGE("Failed to allocate node for %zu pages", BYTES_TO_SIZE_BITS_PAGES(size, page_size_bits));
```
Comment (axel-h) [inline libsel4platsupport/src/io.c:145]: We could do this, then there less redundancy and the asserts become easier to understand:

```
    size_t num_pages = BYTES_TO_SIZE_BITS_PAGES(size, page_size_bits);
    io_mapping_t *mapping = new_node(num_pages);
    if (!mapping) {
        ZF_LOGE("Failed to allocate node for %zu pages", num_pages);
        return NULL;
    }
    assert(mapping->num_pages == num_pages);
```

Comment (Indanz) [reply libsel4platsupport/src/io.c:145]: Agree with using the horrible `BYTES_TO_SIZE_BITS_PAGES` only once.

The assert seems useless, but if you change it to check num_pages, it's really pointless. Either remove it altogether, or leave the check as it is.

Comment (axel-h) [reply libsel4platsupport/src/io.c:145]: Right, it seem useless in the end. I didn't realize that `new_node()` is a local helper function.

Comment (anchao) [reply libsel4platsupport/src/io.c]: Assertion is really useless, let me remove it

File: libsel4platsupport/src/io.c [hunk 2]
```diff
@@ -138,12 +138,13 @@ static void *sel4platsupport_map_paddr_with_page_size(sel4platsupport_io_mapper_
     int page_size = BIT(page_size_bits);
     uintptr_t start = ROUND_DOWN(paddr, page_size);
     uintptr_t offset = paddr - start;
+    size_t num_pages;
     size += offset;
 
-    io_mapping_t *mapping = new_node(BYTES_TO_SIZE_BITS_PAGES(size, page_size_bits));
-    assert(mapping->num_pages << page_size_bits >= size);
+    num_pages = BYTES_TO_SIZE_BITS_PAGES(size, page_size_bits);
```
Comment (axel-h) [inline libsel4platsupport/src/io.c:144]:
```suggestion
    size_t num_pages = BYTES_TO_SIZE_BITS_PAGES(size, page_size_bits);
```

Comment (anchao) [reply libsel4platsupport/src/io.c]: Done

**Review by axel-h — APPROVED** [review]
Looks good to me, thanks.

**Review by Indanz — APPROVED** [review]

**Pull request merged**